test/e2e/integreatly_test.go

```diff
@@ -0,0 +1,141 @@
+	cleanupRetryInterval             = time.Second * 1
+	cleanupTimeout                   = time.Second * 5
+	installationCleanupRetryInterval = time.Second * 20
+	installationCleanupTimeout       = time.Minute * 2 //Longer timeout required to allow for finalizers to execute
+	intlyNamespacePrefix             = "intly-"
+)
+
+func TestIntegreatly(t *testing.T) {
+
+	logf.SetLogger(logf.ZapLogger(true))
+
+	installationList := &operator.InstallationList{}
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, installationList)
+	if err != nil {
+		t.Fatalf("failed to add custom resource scheme to framework: %v", err)
+	}
+	// run subtests
+	t.Run("integreatly", func(t *testing.T) {
+		t.Run("Cluster", IntegreatlyCluster)
+	})
+
+}
+
+func waitForProductDeployment(t *testing.T, f *framework.Framework, ctx *framework.TestCtx, product, deploymentName string) error {
+	namespace := intlyNamespacePrefix + product
+	t.Logf("Checking %s:%s", namespace, deploymentName)
+
+	start := time.Now()
+	err := e2eutil.WaitForDeployment(t, f.KubeClient, namespace, deploymentName, 1, deploymentRetryInterval, deploymentTimeout)
+	if err != nil {
+		return err
+	}
+
+	end := time.Now()
+	elapsed := end.Sub(start)
+
+	t.Logf("%s:%s up, waited %d", namespace, deploymentName, elapsed)
+	return nil
+}
+
+func integreatlyWorkshopTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("could not get namespace: %deploymentName", err)
+	}
+
+	// create installation custom resource
+	workshopInstallation := &operator.Installation{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "e2e-workshop-installation",
+			Namespace: namespace,
+		},
+		Spec: operator.InstallationSpec{
+			Type:             "workshop",
+			NamespacePrefix:  intlyNamespacePrefix,
+			RoutingSubdomain: "apps.example.com",
+			MasterURL:        "http://console.apps.example.com",
+			SelfSignedCerts:  true,
+		},
+	}
+	// use TestCtx's create helper to create the object and add a cleanup function for the new object
+	err = f.Client.Create(goctx.TODO(), workshopInstallation, &framework.CleanupOptions{TestContext: ctx, Timeout: installationCleanupTimeout, RetryInterval: installationCleanupRetryInterval})
+	if err != nil {
+		return err
+	}
+
+	//Auth Stage - verify operators deploy
+	err = waitForProductDeployment(t, f, ctx, "rhsso", "keycloak-operator")
+	if err != nil {
+		return err
+	}
+	//Product Stage - verify operators deploy
+	products := map[string]string{
+		"3scale":               "3scale-operator",
+		"amq-online":           "enmasse-operator",
+		"amq-streams":          "amq-streams-cluster-operator",
+		"codeready-workspaces": "codeready-operator",
+		"fuse":                 "syndesis-operator",
+		"launcher":             "launcher-operator",
+		"nexus":                "nexus-operator",
+		"solution-explorer":    "tutorial-web-app-operator",
+		"user-sso":             "keycloak-operator",
+	}
```

Comment:
Since https://github.com/integr8ly/integreatly-operator/pull/127, this should be checking UPS also, but this hasn't been merged yet https://github.com/integr8ly/manifests/pull/70, so will likely fail.

@maleck13 @philbrookes What is the dev flow you expect when adding or updating products? I'd expect master here to always work against the integreatly app registry, but if we don't merge back manifest changes for product updates first potentially it won't, at least without pulling from a different registry.

I'd also like to update these tests to fail early if tests are missing (i.e. We install ups in a workshop, but don't have a test for it here), but we would end up in dead lock if we expect to merge manifest PRs after the operator one. Should product  update manifest PRs be merged ahead of any integreatly operator PR maybe?

Reply:
What about the possible namespace prefix? We usually use `openshift-` AFAIK.

Reply:
> What about the possible namespace prefix? We usually use openshift- AFAIK.
@tremes Prefix gets used elsewhere as required, this is just listing the products.

Reply:
I think it is fine to merge manifest changes first. followed by the operator. Ideally they should be merged pretty much at the same time